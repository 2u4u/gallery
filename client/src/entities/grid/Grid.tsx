import { FileType } from 'shared/model/reducers/appReducer';
import './index.scss';

type Props = {
    file: FileType;
    handleOpenModal: () => void;
}

function Grid({ file, handleOpenModal }: Props) {
    let content = null;

    switch (file.type) {
        case 'image':
            content = (
                <div className='grid grid--image' style={{ backgroundImage: 'url(' + file.url + ')'}} onClick={handleOpenModal}>
                    <div className="grid__content"></div>
                </div>
            );
            break;
        case 'audio':
        case 'video':
            content = (
                <div className='grid grid--file' onClick={handleOpenModal}>
                    <div className="grid__content">
                        <div className='grid__block'>
                            <div className='grid__align'>
                                <img alt={file.name} className='grid__img' src={`./img/${file.type}.svg`} />
                            </div>
                        </div>
                    </div>
                    <div className='grid__duration'>{file.duration}</div>
                </div>
            )
            break;
        default:
            content = (
                <div className='grid grid--file'>
                    <div className="grid__content">
                        <div className='grid__block'>
                            <div className='grid__align'>
                                <img alt='not defined' className='grid__img' src={`./img/unknown.svg`} />
                            </div>
                        </div>
                    </div>
                </div>
            )
            break;
    }
    
    return (content);
}

export default Grid;
