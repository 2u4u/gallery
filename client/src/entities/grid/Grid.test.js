import renderer from 'react-test-renderer';
import Grid from './Grid';

describe('Grid component ', () => {
    it('matches snapshot', () => {
        const file = {
            _id: 'someid',
            type: 'image',
            name: 'test image',
            url: 'https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png',
            duration: 'undefined',
            date: Date,
        }
        const component = renderer.create(
            <Grid file={file} handleOpenModal={() => false}/>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
