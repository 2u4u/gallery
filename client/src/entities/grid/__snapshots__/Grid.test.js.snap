// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component  matches snapshot 1`] = `
<div
  className="grid grid--image"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png)",
    }
  }
>
  <div
    className="grid__content"
  />
</div>
`;
